zebra-state/src/service/check.rs

```diff
@@ -30,8 +30,7 @@ pub(crate) mod difficulty;
 /// (`POW_AVERAGING_WINDOW + POW_MEDIAN_BLOCK_SPAN`) blocks.
 #[tracing::instrument(
     name = "contextual_validation",
-    fields(?network),
-    skip(prepared, network, finalized_tip_height, relevant_chain)
+    skip(prepared, finalized_tip_height, relevant_chain)
 )]
 pub(crate) fn block_is_contextually_valid<C>(
```

Comment:
1. Confusing names: Let's rename `block_is_contextually_valid` and `block_is_contextually_valid_for_chain` to make it clear how they are different.

2. Confusing API - add to fn doc comment: This function performs checks that require a small number of recent blocks, including previous hash, previous height, and block difficulty.

```suggestion
pub(crate) fn block_is_valid_for_recent_chain<C>(
```

Reply:
Function name changed in 030240f9
Comment added in e416ef68